src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs

```diff
@@ -128,70 +128,57 @@ private static void ReadCallback(UvStreamHandle handle, int status, object state
 
         private void OnRead(UvStreamHandle handle, int status)
         {
-            var normalRead = status >= 0;
-            var normalDone = status == LibuvConstants.EOF;
-            var errorDone = !(normalDone || normalRead);
-            var readCount = normalRead ? status : 0;
-
-            if (normalRead)
+            if (status >= 0)
```

Comment:
It its but I think we should just avoid signaling the reader if nothing was written.